lib/iruby/ostream.rb

```diff
@@ -19,32 +19,44 @@ def isatty
     end
     alias_method :tty?, :isatty
 
-    def read(*args)
+    def read(*_args)
       raise IOError, 'not opened for reading'
     end
     alias_method :next, :read
     alias_method :readline, :read
 
-    def write(s)
-      raise 'I/O operation on closed file' unless @session
-      @session.send(:publish, :stream, name: @name, text: s.to_s)
-      nil
+    def write(*obj)
+      str = building_string { |sio| sio.write(*obj) }
+      session_send(str)
     end
     alias_method :<<, :write
     alias_method :print, :write
 
-    def printf(*fmt)
-      write sprintf(*fmt)
+    def printf(format, *obj)
+      str = building_string { |sio| sio.printf(format, *obj) }
+      session_send(str)
     end
 
-    def puts(*lines)
-      lines = [''] if lines.empty?
-      lines.each { |s| write("#{s}\n")}
-      nil
+    def puts(*obj)
+      str = building_string { |sio| sio.puts(*obj) }
+      session_send(str)
     end
 
     def writelines(lines)
       lines.each { |s| write(s) }
     end
+
+    private
+
+    def building_string
```

Comment:
I think a gerund is not fit to the method name.

Reply:
Hmm. That makes sense..